docassemble/ALAutomatedTestingTests/al_set_up_testing.py

```diff
@@ -138,6 +164,39 @@ def make_pull_request( self ):
     
     return self
   
+  def create_secrets( self ):
+    """Set the GitHub repo secrets the tests need to log into the da server and
+    create projects to contain the interviews being tested. PyGithub does not yet handle secrets. See issue: https://github.com/PyGithub/PyGithub/issues/1373."""
+    self.put_secret( 'PLAYGROUND_EMAIL', self.email )
+    self.put_secret( 'PLAYGROUND_PASSWORD', self.password )
+    self.put_secret( 'PLAYGROUND_ID', self.playground_id )
+    return self
+  
+  def put_secret( self, name, value ):
+    """Add or update one secret to the GitHub repo."""
+    # Create repo secret: https://docs.github.com/en/rest/reference/actions#create-or-update-a-repository-secret
+    # Convert the message and key to Uint8Array's (Buffer implements that interface)
+    encrypted_key = public.PublicKey( self.public_key.encode("utf-8"), encoding.Base64Encoder() )
+    sealed_box = public.SealedBox( encrypted_key )  # ?
+    encrypted = sealed_box.encrypt( value.encode( "utf-8" ))  # LibSodium
+    base64_encrypted = b64encode( encrypted ).decode( "utf-8" )  # turns into string
+    
+    url = self.github_repo_base + "/actions/secrets/" + name
+    payload = '{"encrypted_value":"' + base64_encrypted + '", "key_id":"' + self.key_id + '"}'
```

Comment:
This might be easier to read with an f-string. Something like 
```
>>> name = "Eric"
>>> age = 74
>>> f"Hello, {name}. You are {age}."
```
Source: https://realpython.com/python-f-strings/

Reply:
Or creating a regular dict and then `json.dumps`ing it to turn it into a string.

Reply:
I have not seen that way of composing strings in python before! All the ways I've seen it done have been incredibly clunky. This is much better.

I really like the idea of `json.dumps` too. It seems scalable. I'd seen it around, but wasn't sure of its use and didn't take the time to find out.

Which do you think would be a better fit?

Reply:
I think `json.dumps` would be a better fit here. That seems more in line with what we're trying to do: create a dictionary payload that is represented as a string.